apps/spacer/spacer.app.js

```diff
@@ -0,0 +1,210 @@
+/* -> space race ! */
+
+/* 
+
+apps/assistedgps/custom.html
+
+https://github.com/espruino/EspruinoDocs/blob/master/info/Bangle.js2%20Technical.md#gps
+
+gsa mi rika 2d/3d fix, a taky pdop/vdop/hdop
+
+CFG-NAVX z CASIC_en -- umoznuje nastavit chodec / auto / letadlo
+
+MON-VER -- vrati version stringy
+
+CFG-PMS z gpssetup to zda se neumi?
+
+2.11.5 CFG-RATE (0x06 0x04)
+*/
+
+/* ui library 0.1.3 */
+let ui = {
+  display: 0,
+  numScreens: 2,
+  drawMsg: function(msg) {
+    g.reset().setFont("Vector", 35)
+      .setColor(1, 1, 1)
+      .fillRect(0, this.wi, this.w, this.y2)
+      .setColor(0, 0, 0)
+      .drawString(msg, 5, 30)
+      .flip();
+  },
+  drawBusy: function() {
+    this.drawMsg("\n.oO busy");
+  },
+  nextScreen: function() {
+    print("nextS");
+    this.display = this.display + 1;
+    if (this.display == this.numScreens)
+      this.display = 0;
+    this.drawBusy();
+  },
+  prevScreen: function() {
+    print("prevS");
+    this.display = this.display - 1;
+    if (this.display < 0)
+      this.display = this.numScreens - 1;
+    this.drawBusy();
+  },
+  onSwipe: function(dir) {
+    this.nextScreen();
+  },
+  wi: 24,
+  y2: 176,
+  h: 152,
+  w: 176,
+  last_b: 0,
+  topLeft: function() { this.drawMsg("Unimpl"); },
+  topRight: function() { this.drawMsg("Unimpl"); },
+  touchHandler: function(d) {
+    let x = Math.floor(d.x);
+    let y = Math.floor(d.y);
+    
+    if (d.b != 1 || this.last_b != 0) {
+      this.last_b = d.b;
+      return;
+    }
+    
+    print("touch", x, y, this.h, this.w);
+
+    if ((x<this.w/2) && (y<this.y2/2))
+      this.topLeft();
+    if ((x>this.w/2) && (y<this.y2/2))
+      this.topRight();
+    if ((x<this.w/2) && (y>this.y2/2)) {
+      print("prev");
+      this.prevScreen();
+    }
+    if ((x>this.w/2) && (y>this.y2/2)) {
+      print("next");
+      this.nextScreen();
+    }
+  },
+  init: function() {
+    this.h = this.y2 - this.wi;
+    this.drawBusy();
+  }
+};
+
+let graw = {
+  old_msg: {},
+  msg: {},
+  tof: function(v) { let i = (1*v); return i.toFixed(0); },
+  fmtSys: function(sys) {
+    return sys.sent + "." + sys.d23 + "D "+ this.tof(sys.pdop) + " " + this.tof(sys.vdop) + "\n";
+  },
+  display: function() {
+    let m = this.old_msg;
+    let msg = "" + this.tof(m.time) + "\n" + 
+        "q" + m.quality + " " + m.in_view + " " + m.hdop + "\n" +
+        "gp"+ this.fmtSys(m.gp) +
+        "bd" + this.fmtSys(m.bd)  +
+        "gl" + this.fmtSys(m.gl);
```

Comment:
These cause an exception the first time round, looks like (at least) `gp` isn't defined:

```
Uncaught Error: Can't read property 'sent' of undefined
 at line 94 col 15 in spacer.app.js
sys.sent + "." + sys.d23 + "D "+ this.tof(sys.pdop) + " " + ...
   ^
in function "fmtSys" called from line 100 col 31 in spacer.app.js
        "gp"+ this.fmtSys(m.gp) +
                              ^
in function "display" called from line 200 col 41 in spacer.app.js
graw.display();
             ^
```